src/datastore_mad/remotes/lxd_downloader.sh

```diff
@@ -0,0 +1,161 @@
+        version="Stretch"
+    elif [[ $version_number == "8" ]]; then
+        version="Jessie"
+    else
+        version="undefined"
+    fi
+    release_date=""
+elif [[ $rootfs_url == *"linuxcontainers.org"* ]]; then
+    untar_options="xvJpf"
+    extension="tar.xz"
+    distro=`echo $rootfs_url | cut -d '/' -f 5`
+    version=`echo $rootfs_url | cut -d '/' -f 6`
+    release_date=`echo $rootfs_url | cut -d '/' -f 9`
+fi
+
+output=$tmp_dir/$id.$extension
+
+$curl $rootfs_url --output $output --silent
+qemu-img create -f raw $tmp_dir/$id.raw ${size}M  > /dev/null 2>&1
+
+case $filesystem in
+    "ext4")
+        mkfs.ext4 -F $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+    "xfs")
+        mkfs.xfs -f $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+    *)
+        mkfs.ext4 -F $tmp_dir/$id.raw > /dev/null 2>&1
+        ;;
+esac
+
+
+mkdir $tmp_dir/$id
+sudo mount $tmp_dir/$id.raw $tmp_dir/$id
+sudo chown oneadmin:oneadmin $tmp_dir/$id
+sudo tar $untar_options $output -C $tmp_dir/$id > /dev/null 2>&1
+sync
+
+context_url="https://github.com/OpenNebula/addon-context-linux/releases/download"
+
+case "$rootfs_url" in
+*ubuntu*|*debian*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        export PATH=$PATH:/bin:/sbin
+        rm -f /etc/resolv.conf > /dev/null 2>&1
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        apt-get update > /dev/null
+        apt-get install curl -y > /dev/null 2>&1
+        $curl $context_url/v$selected_tag/one-context_$selected_tag-1.deb -Lsfo /root/context.deb
+        apt-get install /root/context.deb -y > /dev/null 2>&1
+EOT
+    ;;
+*centos/6*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        export PATH=$PATH:/bin:/sbin
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el6.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
+EOT
+    ;;
+*centos/7*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el7.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
+EOT
+    ;;
+*alpine*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        echo "nameserver $dns_server" > /etc/resolv.conf
+        $curl $context_url/v$selected_tag/one-context-$selected_tag-1.el7.noarch.rpm -Lsfo /root/context.rpm
+        yum install /root/context.rpm -y > /dev/null 2>&1
+EOT
+    ;;
+*opensuse*)
+    terminal="/bin/ash"
+    read -r -d '' commands << EOT
+        echo "OpenSuse is not yet supported" > /root/opennebula_context.log
+EOT
+    ;;
+*)
+    terminal="/bin/bash"
+    read -r -d '' commands << EOT
+        echo "This distro is not supported by OpenNebula context" > /root/opennebula_context.log
+EOT
+    ;;
+esac
+
+cat << EOF | sudo chroot $tmp_dir/$id $terminal
```

Comment:
Can't be used `fakeroot` somehow? Enable sudo chroot is big security issue.

Reply:
Tried but couldn't. Containers from linuxcontainers have root as owner and group for most files insode .tar.xz. When uncompressing, I need to respect ownership (I can't simply massively change ownership to everything). Because everything belongs to root, fakeroot won't work.